### PR TITLE
kraken: selinux: Do parallel relabel on package install

### PR DESCRIPTION
**Pull request by b-ranto**
http://tracker.ceph.com/issues/20191
http://tracker.ceph.com/issues/20193

This backports the following PRs for ceph-disk fix related bits (making parallel relabel on package install and a packaging bug fix):

https://github.com/ceph/ceph/pull/14871
https://github.com/ceph/ceph/pull/15490

### CODE REVIEW

**Comment by b-ranto** [comment]
jenkins retest this please

**Comment by b-ranto** [comment]
jenkins retest this please

**Comment by smithfarm** [comment]
Jenkins re-test this please

File: ceph.spec.in [hunk 1]
```diff
@@ -654,6 +654,7 @@ Summary:	SELinux support for Ceph MON, OSD and MDS
 Group:		System Environment/Base
 Requires:	ceph-base = %{epoch}:%{version}-%{release}
 Requires:	policycoreutils, libselinux-utils
+Requires(post):	ceph-base = %{_epoch_prefix}%{version}-%{release}
```
Comment (smithfarm) [inline ceph.spec.in]: @b-ranto: There is no `%_epoch_prefix` in the kraken spec file

**Comment by b-ranto** [comment]
The fix for #16208 is in, now.

File: ceph.spec.in (hunk 1 above)
Comment (smithfarm) [inline ceph.spec.in]: @b-ranto: Need to fix this, too.

**Comment by smithfarm** [comment]
Marking DNM again because it breaks the rados suite: 

```
2017-07-07T19:57:18.547 INFO:teuthology.orchestra.run.smithi146.stderr:Error: Package: 1:ceph-selinux-11.2.0-697.ga7d2f1d.el7.x86_64 (ceph)
2017-07-07T19:57:18.552 INFO:teuthology.orchestra.run.smithi146.stderr:           Requires: ceph-base = %{_epoch_prefix}11.2.0-697.ga7d2f1d.el7
2017-07-07T19:57:18.561 INFO:teuthology.orchestra.run.smithi146.stderr:           Available: 1:ceph-base-11.2.0-697.ga7d2f1d.el7.x86_64 (ceph)
2017-07-07T19:57:18.566 INFO:teuthology.orchestra.run.smithi146.stderr:               ceph-base = 1:11.2.0-697.ga7d2f1d.el7
```

**Comment by b-ranto** [comment]
The _epoch_prefix should be fixed in the latest patch iteration.

File: ceph.spec.in [hunk 2]
```diff
@@ -654,6 +654,7 @@ Summary:	SELinux support for Ceph MON, OSD and MDS
 Group:		System Environment/Base
 Requires:	ceph-base = %{epoch}:%{version}-%{release}
 Requires:	policycoreutils, libselinux-utils
+Requires(post):	ceph-base = %{epoch}%{version}-%{release}
```
Comment (smithfarm) [inline ceph.spec.in]: missing a `:` between epoch and version macros

**Comment by smithfarm** [comment]
Marking DNM because it breaks the rados suite

**Comment by b-ranto** [comment]
should be fixed, now

**Comment by smithfarm** [comment]
This PR is now in the integration branch.

**Comment by smithfarm** [comment]
@b-ranto This no longer breaks the rados suite, but it appears to break the ceph-disk suite:

http://pulpito.ceph.com/smithfarm-2017-08-01_09:33:50-ceph-disk-wip-kraken-backports-distro-basic-smithi/

Please take a look.

**Comment by smithfarm** [comment]
Compare with http://pulpito.ceph.com/teuthology-2017-07-29_05:10:02-ceph-disk-kraken-distro-basic-vps/

**Comment by smithfarm** [comment]
@tchaikov @dachary This passed a ceph-disk suite at http://tracker.ceph.com/issues/19009#note-77

I seem to remember that the ceph-disk suite only runs on vps? It fails on smithi - see http://pulpito.ceph.com/smithfarm-2017-08-01_09:33:50-ceph-disk-wip-kraken-backports-distro-basic-smithi/ - but since it succeeds on vps I'm guessing the smithi run can be disregarded.

@tchaikov already approved this, but it would be nice if one of you could put my mind to rest concerning the failed smithi run.

This also passed rados and upgrade/jewel-x btw.

**Comment by b-ranto** [comment]
@smithfarm There are no teuthology tests for this part of the ceph-disk code. The code is only tested by the ceph-disk tests as part of the make check run.

**Comment by tchaikov** [comment]
> I seem to remember that the ceph-disk suite only runs on vps? 

yes, @dachary reminded me on this when i reported that the ceph-disk test suite fails on mira/smithi nodes.

**Pull request merged**